frame/vamm/src/lib.rs

```diff
@@ -341,7 +411,121 @@ pub mod pallet {
 			})
 		}
 
-		fn get_twap(vamm_id: &Self::VammId) -> Result<Self::Decimal, DispatchError> {
+		/// Gets the current price of the __base__ or __quote__ asset in a vamm.
+		///
+		/// # Overview
+		/// In order for the caller knows which is the current price of an asset
+		/// in an specific vamm it has to request it to the Vamm Pallet. The
+		/// Vamm Pallet consults the runtime storage for the desired vamm,
+		/// computes the current price and returns it to the caller.
+		///
+		/// ![](https://www.plantuml.com/plantuml/svg/PP0zJWCn44PxdsBO1b2q5qY14b9GKI7H3vkFOB7-OURRvFfWhm0XEillpHlBEwSQbpG7Vu-vgcaIWzUI7OzmrnFkCPVBtgnSXBOWC7A6F82Yxg1KYnFajPYeF6jAuLeN5fqOpqf8oU6ARqYGfEOXL3N6ALRDbE4mHsGEeYvJF_x5BTVXkNMFIdrHXmnFBAOdo4qJRhlXNGbhHSQxFhBPRFyzrF2nm1aQRruVNBL-vLJYXwxmK59TY5xuPbzmNJQEMzd_BWWxv6Fxq4y0)
+		///
+		/// ## Parameters
+		///  - `vamm_id`: The ID of the desired vamm to query.
+		///  - `asset_type`: The desired asset type to get info about. (either
+		///  __base__ or __quote__)
+		///
+		/// ## Returns
+		/// The price of __base__ asset in relation to __quote__
+		/// (or vice-versa).
+		///
+		/// ## Assumptions or Requirements
+		/// In order to consult the current price for an asset, we need to
+		/// ensure that the desired vamm_id exists.
+		///
+		/// ## Emits
+		/// No event is emitted for this function.
+		///
+		/// ## State Changes
+		/// This function does not mutate runtime storage.
+		///
+		/// ## Errors
+		/// * [`Error::<T>::VammDoesNotExist`]
+		/// * [`Error::<T>::FailToRetrieveVamm`]
+		/// * [`ArithmeticError::Overflow`](sp_runtime::ArithmeticError)
+		/// * [`ArithmeticError::DivisionByZero`](sp_runtime::ArithmeticError)
+		///
+		/// # Runtime
+		/// `O(1)`
+		fn get_price(
+			vamm_id: VammIdOf<T>,
+			asset_type: AssetType,
+		) -> Result<BalanceOf<T>, DispatchError> {
+			// Requested vamm must exist.
+			ensure!(VammMap::<T>::contains_key(vamm_id), Error::<T>::VammDoesNotExist);
+
+			let vamm_state = VammMap::<T>::get(vamm_id).ok_or(Error::<T>::FailToRetrieveVamm)?;
```

Comment:
Is it possible to merge these two?
```suggestion
			let vamm_state = VammMap::<T>::get(vamm_id).ok_or(Error::<T>::VammDoesNotExist)?;
```
It would imply removing either `VammDoesNotExist` or `FailToRetrieveVamm`...

Reply:
Yeah, we could do that but I purposefully chose to use the `ensure!` command because in essence we could get two different errors in this `get` call. 
(Well at least I think those are two different errors, and would be worth to notify the caller about them)

What do you think?

Reply:
I'd have to check if `get` can fail for any reason other than the `vamm_id` not existing. If that's not the case, then I think it's fair to have only one error